backend/views/views.py

```diff
@@ -86,13 +86,39 @@ def user_modify(users_id):
         return render_template('user_modify.html', form=form, route_to=route_to)
 
 
-@app.route('/deleteuser/<int:users_id>')
+@app.route('/deleteuser/<int:users_id>', methods=['POST'])
+@admin_permissions
 def delete_user(users_id):
+    msg = "cannot delete user"
     current_moment = datetime.now()
     user = db.session.query(User).get(users_id)
-    user.delete_date = current_moment
+    if not is_last_admin(user):
+        user.delete_date = current_moment
```

Comment:
As we agreed before, better to add method to `Users` model, which performs all business logic to delete user. And one more thing, please do not use `datetime.now()` as value for `delete_date` field, as it uses time on web server side. Use [`functions.current_timestamp`](http://docs.sqlalchemy.org/en/latest/core/functions.html#sqlalchemy.sql.functions.current_timestamp), as it uses current timestamp on DB server side.